alacritty/src/event.rs

```diff
@@ -381,16 +384,54 @@ impl<N: Notify + OnResize> Processor<N> {
             config,
             message_buffer,
             display,
+            event_queue: Vec::new(),
         }
     }
 
+    /// Return `true` if `event_queue` is empty, `false` otherwise.
+    #[inline]
+    #[cfg(not(any(target_os = "macos", windows)))]
+    fn event_queue_empty(&mut self) -> bool {
+        let wayland_event_queue = match self.display.wayland_event_queue.as_mut() {
+            Some(wayland_event_queue) => wayland_event_queue,
+            None => return self.event_queue.is_empty(),
+        };
+
+        // On Wayland we also dispatch and check its own event queue
```

Comment:
The `also` sounds a bit strange here, since there's nothing previously that it could refer to. I'd suggest the following:

```suggestion
        // Check for pending frame callbacks.
```

Alternatively:

```suggestion
        // Check for pending frame callbacks on Wayland.
```